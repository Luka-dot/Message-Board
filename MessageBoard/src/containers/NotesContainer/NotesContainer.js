import React, { Component } from 'react';
import NoteDisplay from '../../components/NoteDisplay/NoteDisplay';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import Masonry from 'react-masonry-css'

export class NotesContainer extends Component {

  generateNotes = () => {
    return this.props.allNotes.map(note =>
      <Link to={`/notes/${note.id}`} key={note.id} className='note-click' style={{ backgroundColor: note.background }}>
        <NoteDisplay {...note} />
      </Link>
    )
  }

  render() {
    const { allNotes } = this.props;
    const loader = this.props.loading ? <div></div> : null
    let splashMessage = null;
    if (!loader && !allNotes.length) splashMessage = <h1 className="splash-message">Try Creating a New Note!</h1>;
    const breakpointColumnsObj = {
      default: 4,
      1100: 3,
      700: 2,
      500: 1
    };
    return (
      <div className="NotesContainer">
        { loader }
        { splashMessage }
        <Masonry
          breakpointCols={breakpointColumnsObj}
          className="my-masonry-grid"
          columnClassName="my-masonry-grid_column">
          { this.props.allNotes && this.generateNotes() }
        </Masonry>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  allNotes: state.allNotes,
  loading: state.loading
})

NotesContainer.propTypes = {
  allNotes: PropTypes.array,
  loading: PropTypes.bool,
}

export default connect(mapStateToProps)(NotesContainer);
