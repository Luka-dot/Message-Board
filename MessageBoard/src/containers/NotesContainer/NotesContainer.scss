.NotesContainer {

  .note-click {
    border-radius: 15px;
    background: #ffe680;
    text-decoration: none;
    margin: 1em;
    padding: 1em;
    display: flex;
    font-family: $font-Abel;
    box-shadow: 4px 6px 6px 3px #00000059;
    position: relative;

    @media screen and (max-width: 787px) {
      margin: .5em;
    }

    @media screen and (max-width: 500px) {
      margin: auto;
      margin-top: 1em;
      margin-bottom: 1em;
      width: 80%;
    }

    &:before {
      content:"";
      background-image: url('./media/tack.png');
      position: absolute;
      width:100px;
      height: 100px;
      top: .2em;
      right: 3.5em;
      background-repeat: no-repeat;
    }
  }

  .splash-message {
    margin: 100px auto 0;
    color: #ffffff;
    font-size: 80px;
    width: 500px;
    text-shadow: 2px 4px 13px #0000006e;
  }

  .my-masonry-grid {
    display: flex;
    width: auto;
  }

  .my-masonry-grid_column {
    background-clip: padding-box;
  }

}
