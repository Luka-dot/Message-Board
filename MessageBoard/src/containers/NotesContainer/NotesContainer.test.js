import React from 'react';
import { shallow } from 'enzyme';
import { NotesContainer, mapStateToProps } from './NotesContainer';

const mockLoading = false;

describe('NotesContainer', () => {
  describe('NotesContainer', () => {
    let wrapper;

    const mockShortid = jest.fn().mockReturnValue('12345789');

    const mockAllNotes = [
        { id: "123ABA",
        title: 'Worf ToDo',
        list: [
          { id: mockShortid,
            text: 'Eat food',
            isComplete: false
          },
        ]
      },
      { id: "987GDGFD",
      title: 'Jake ToDo',
      list: [
        { id: mockShortid,
          text: 'Do basic styling',
          isComplete: false
        },
      ]
    }];

  beforeEach(() => {
    wrapper = shallow(<NotesContainer allNotes={mockAllNotes} loading={mockLoading}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should return JSX when generateNotes is called', () => {
    expect(wrapper.generateNotes).toMatchSnapshot();
  })

  it('should match the snapshot when loading is true', () => {
    wrapper = shallow(<NotesContainer allNotes={[]} loading={false}/>)
    expect(wrapper).toMatchSnapshot();
  })


});

  describe('mapStateToProps', () => {
    it('should return an object with a notes array', () => {
      const mockState = {
        allNotes: [
          { id: "123ABA",
            title: 'Worf ToDo',
            list: [
              { id: "123", text: 'Eat food', isComplete: false },
            ]
          },
          { id: "987GDGFD",
            title: 'Jake ToDo',
            list: [
              { id: "0098", text: 'Do styling', isComplete: false },
            ]
          }
        ],
        error: "",
        loading: false
      }
      const expected = {
        allNotes: [
          { id: "123ABA",
            title: 'Worf ToDo',
            list: [
              { id: "123", text: 'Eat food', isComplete: false },
            ]
          },
          { id: "987GDGFD",
            title: 'Jake ToDo',
            list: [
              { id: "0098", text: 'Do styling', isComplete: false },
            ]
          }
        ],
        loading: false
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})
