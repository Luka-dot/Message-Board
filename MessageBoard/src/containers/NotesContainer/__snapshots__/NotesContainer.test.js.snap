// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContainer NotesContainer should match the snapshot 1`] = `
<div
  className="NotesContainer"
>
  <Masonry
    breakpointCols={
      Object {
        "1100": 3,
        "500": 1,
        "700": 2,
        "default": 4,
      }
    }
    className="my-masonry-grid"
    columnClassName="my-masonry-grid_column"
  >
    <Link
      className="note-click"
      key="123ABA"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      to="/notes/123ABA"
    >
      <NoteDisplay
        id="123ABA"
        list={
          Array [
            Object {
              "id": [MockFunction],
              "isComplete": false,
              "text": "Eat food",
            },
          ]
        }
        title="Worf ToDo"
      />
    </Link>
    <Link
      className="note-click"
      key="987GDGFD"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      to="/notes/987GDGFD"
    >
      <NoteDisplay
        id="987GDGFD"
        list={
          Array [
            Object {
              "id": [MockFunction],
              "isComplete": false,
              "text": "Do basic styling",
            },
          ]
        }
        title="Jake ToDo"
      />
    </Link>
  </Masonry>
</div>
`;

exports[`NotesContainer NotesContainer should match the snapshot when loading is true 1`] = `
<div
  className="NotesContainer"
>
  <h1
    className="splash-message"
  >
    Try Creating a New Note!
  </h1>
  <Masonry
    breakpointCols={
      Object {
        "1100": 3,
        "500": 1,
        "700": 2,
        "default": 4,
      }
    }
    className="my-masonry-grid"
    columnClassName="my-masonry-grid_column"
  />
</div>
`;

exports[`NotesContainer NotesContainer should return JSX when generateNotes is called 1`] = `undefined`;
